ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializerTest.java

```diff
@@ -391,4 +392,38 @@ public void shouldTransformSourceNameDelimiterForInternal() {
     final GenericRow deserializedKsqlRecord = serde.deserializer().deserialize("topic", bytes);
     assertThat(deserializedKsqlRecord, equalTo(ksqlRecord));
   }
+
+    @Test
+    public void shouldUseSchemaNameFromPropertyIfExists() {
+        final String schemaName = "TestSchemaName1";
+        final String schemaNamespace = "com.test.namespace";
+
+        final Schema ksqlSchema = Schema.OPTIONAL_STRING_SCHEMA;
+        final Object ksqlValue = "foobar";
+
+        final Map<String, String> properties = new HashMap<>();
+        properties.put(KsqlAvroTopicSerDe.AVRO_SCHEMA_FULL_NAME,
+                       String.join(".", schemaNamespace, schemaName));
+
+        final Schema ksqlRecordSchema = SchemaBuilder.struct()
+            .field("field0", ksqlSchema)
+            .build();
+
+        final GenericRow ksqlRecord = new GenericRow(ImmutableList.of(ksqlValue));
+
+        final SchemaRegistryClient schemaRegistryClient = new MockSchemaRegistryClient();
```

Comment:
nit: would you mind pulling this out into a field, (same for other tests if you have time).